crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs

```diff
@@ -282,6 +287,33 @@ impl OffChainDatabase for OffChainIterableKeyValueView {
                 .into_boxed()
         }
     }
+
+    fn coins_to_spend_index(
+        &self,
+        owner: &Address,
+        asset_id: &AssetId,
+    ) -> CoinsToSpendIndexIter {
+        let prefix: Vec<_> = NON_RETRYABLE_BYTE
```

Comment:
Maybe you want to return an error if indexation is disabled=)

Reply:
Yeah, that could come in handy, we just don't have this information available here. We would need to query metadata for every request or stick a boolean flag in, hmm... "IterableKeyValueView<ColumnType<OffChain>>"?

Am I maybe missing something?

Reply:
It is true that we need to request it. If each request will do that, it should be fast because the result will be cached. But maybe we want to keep this information inside of the `GenericDatabase` idk and make this check fast=) It can be follow up PR and there we can decide how better to do it

Reply:
Hmm, maybe limit access on trait level? Something like `pub trait OffChainDatabaseWithCoinIndex: OffChainDatabase` - not sure how feasible this is. 

But yeah, I'll create a follow-up issue anyway, because it's not a critical flaw as long as we only call it from the context of `coins_to_spend_with_cache()`

Reply:
Follow-up issue: https://github.com/FuelLabs/fuel-core/issues/2499